code/modules/food_and_drink/ingredients.dm

```diff
@@ -582,41 +582,45 @@
 	var/toppingstext = null
 
 	attackby(obj/item/W as obj, mob/user as mob)
-		if (istype(W, /obj/item/reagent_containers/food/snacks/))
-			var/obj/item/reagent_containers/food/snacks/F = W
-			if(!F.custom_food)
-				return
+		if (istype(W, /obj/item/reagent_containers/food))
+			add_ingredient(W, user)
+		else
+			return
+	proc/add_ingredient(obj/item/W as obj, mob/user as mob)
+		var/obj/item/reagent_containers/food/snacks/F = W
+		if(!F.custom_food)
```

Comment:
Is `custom_food` a var on `/obj/item`? If not, need to make sure `F` exists before doing anything with it.